README.md

```diff
@@ -287,53 +287,161 @@ Although this example is pretty simple and not even includes Lens composition, i
 
 Just as with React, but assuming Angular instead :)
 
+The code for below example - can be found here <https://github.com/georgebatalinski/telescope_todo>
+
 As a bootstrap for the application we need something like:
 
+```typescript 
+@NgModule({
+  declarations: [
+    AppComponent,
+    TodoListComponent
+  ],
+  imports: [
+    ...
+  ],
+  providers: [
+    { provide: 'StreamContract', useFactory: ()=> Telescope.of( AppState.empty() ), deps: [] }
+  ],
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
+```
+
+We define our global STATE (however, the more ultimate approach would be to have local state)
+local state = would mean each Service, would have a Telescope instance of their own, resulting in us not 
+having to setup and/or maintain <AppState> 
+Note: If choose to maintain <AppState> as in this example, we would be adding more members to the 
+class, which would mean boilerplate. 
+
+
+```typescript
+interface STATE {
+    todos: TodoState;
+    people: PersonsState;
+}
+
+
+export class AppState implements STATE {
+
+    static empty() {
+        return new AppState( 
+            new TodoState(), 
+            new PersonsState() 
+            );
+    }
+    todos: TodoState;
+    people: PersonsState;
+    
+    constructor(todos: TodoState, people: PersonsState ) {
+        this.todos = todos;
+        this.people = people;
+    }
+}
+
+
+export class TodoState {
+    todos: Array<Todo>;
+
+    constructor( todos: Array<Todo> = [] ) {
+        this.todos = todos;
+    }
+
+    add(title: string): TodoState {
+        let todos = [ {title } ];
+        return new TodoState( [...this.todos, ...todos] );
+    }
+}
+
+export class Todo {
+    constructor( public title: string = ''){}
+}
+
+export class PersonsState {
+    people: Array<Person>;
+}
+
+
+class Person {
+    name: string;
+}
+```
+
+Setup a Service - that will use `StreamContract`
+
+```typescript 
+export class TodoService {
+  state$: Telescope<TodoState>;
+  loaded: boolean = false;
+  constructor(@Inject('StreamContract') private streamContract: Telescope<AppState>) { 
+    /*
+      this will allow us to slice the GLOBAL state 
+      so we can only work with TODOs 
+      since in the AppState  has people
+    */
+    this.state$ = this.streamContract.magnify( todosLens() );  
```

Comment:
```
export const todosLens = () =>
  new Lens<AppState, TodoState>(
    (state) => {
        return state.todos;
    },
    (todos, state) => {
        state.todos = todos;
        return state;
    }
  );

```

Reply:
I think `AppState` should be immutable, that is, `todos` and `people` should be `readonly`, that way it is very evident that the changes are fed to the app through the `stream` property of the `telescope`.

That requires you to change your implementation, which can be done easily, I think none of the classes you added need to be a class but rather an interface and then you can always use the spread operator (you already do) for the lenses getters and setters.

Makes sense? That said, and specially given how long this review is taking (on my side) I propose to merge as is and I can make the changes later. I will send a new PR with those changes.